.github/workflows/proto-registry.yml

```diff
@@ -0,0 +1,92 @@
+
+# Provenance publishes Protobuf content (including third_party dependencies) to buf.build.
+# Because of third_party dependencies declared in buf.yml, a buf.lock will be generated
+# containing the dependency manifest. Among other things, the buf.lock contains the
+# latest commit hash of the third_party library in the buf.build Provenance repository.
+# When new files are added to the third_party library we need:
+#   1. publish the third_party library
+#   2. update buf.lock to point to the new commit hash
+#   3. sign and create a PR to commit buf.lock changes back to main
+# This helps avoid breaking changes when we cut new releases.
+jobs:
+  push_third_party:
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Buf setup action
+        uses: bufbuild/buf-setup-action@v1.13.1
+      - name: Buf push 'third_party/proto'
+        uses: bufbuild/buf-push-action@v1
+        with:
+          input: 'third_party/proto'
+          buf_token: ${{ secrets.BUF_TOKEN }}
+      - name: Import GPG key
+        id: import_gpg
+        uses: crazy-max/ghaction-import-gpg@v5
+        with:
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.GPG_PASSPHRASE }}
+          git_user_signingkey: true
+          git_commit_gpgsign: true
+      - name: Update buf.lock
+        run: |
+          cd proto
+          buf mod update
+          cd ..
+          git add .
+          git commit -S -m "Update buf.lock to latest commit hash"
+      - name: Create Pull Request
+        id: cpr
+        uses: peter-evans/create-pull-request@v4.2.3
+        with:
+          base: main
+          branch: create-pull-request/patch-buf-lock
+          delete-branch: true
+          # GitHub Personal Access Token (from the same account where the GPG key is stored)
+          token: ${{ secrets.CPR_PAT }}
+          committer: ${{ steps.import_gpg.outputs.email }}
+          signoff: true
+          title: 'Update buf.lock'
+          body: |
+            Updates `third_party/proto/buf.lock` with latest commit hash.
+            - Auto-generated by [create-pull-request][1]
+
+            [1]: https://github.com/peter-evans/create-pull-request
+      - name: Check outputs
+        if: ${{ steps.cpr.outputs.pull-request-number }}
+        run: |
+          echo "Pull Request Number - ${{ steps.cpr.outputs.pull-request-number }}"
+          echo "Pull Request URL - ${{ steps.cpr.outputs.pull-request-url }}"
+
+  push_proto:
+    if: github_ref_type == 'tag'
+    runs-on: ubuntu-latest
+    name: Protobuf Push
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Buf setup action
+        uses: bufbuild/buf-setup-action@v1.13.1
+      - name: Buf push 'proto/'
+        uses: bufbuild/buf-push-action@v1
+        with:
+          input: 'proto'
+          buf_token: ${{ secrets.BUF_TOKEN }}
```

Comment:
Since we should only push our protos during a release, this should really be in the `release.yaml`. That way A) we don't have to duplicate logic, and B) its in the chart when looking at the release Action in Github.

Flow:
1. The third-party protos get pushed whenever something is merged to `main`.
2. That creates a PR to update the lock.
3. That PR gets merged.
4. Things get back-ported to a release branch.
5. We tag the release.
6. Our protos get pushed, which reference the third party ones that were previously pushed and so are up-to-date.

We should take this job out of this workflow. Then, in the releases workflow, restore the `buf_push` job, but remove the step that pushes the third party protos.

Reply:
I'll put it back. 

I wasn't quite sure if it be better keeping the BSR push together even though they happened on different triggers (main, release).

Reply:
`proto_push` restored in `release.yml`